### DISCUSSION TITLE
Flask socket io returning session invalid

### DISCUSSION THREAD

**Discussion by PrimeTheFirst** [miguelgrinberg/Flask-SocketIO#1935] [Q&A]
I am making an app which incorporates chat. To naturally I used flask socket io. In development the chat worked , but when I put it in production, it stopped and gave me the error
` http://rcampusbuddy.bumblr.in/socket.io/?EIO=4&transport=polling&t=ONnqHU8&sid=ZzGCvCBRFp5YotGFAAAQ
` 
When I clicked this error, it says session invalid. According to some stackoverflow posts, it can happen due to multiple things like gevent eventlet. But I have not installed any one of these. 
In html client side I have used:
` <script src="https://cdnjs.cloudflare.com/ajax/libs/socket.io/4.5.4/socket.io.js" integrity="sha512-YeeA/Qxn5hYdkukScTCNNOhTrv1C2RubAGButJ1rmgQwZf/HdRaCGl+JAVkqsqaNRaYNHdheiuKKuPf9mDcqKg==" crossorigin="anonymous" referrerpolicy="no-referrer">`

In python I have the following libraries installed:

1. bcrypt==4.0.1
2. bidict==0.22.1
3. click==8.1.3
4. dnspython==2.3.0
5. eventlet==0.33.3
6. Flask==2.2.2
7. Flask-SocketIO==5.2.0
8. greenlet==2.0.1
9. itsdangerous==2.1.2
10. Jinja2==3.1.2
11. MarkupSafe==2.1.2
12. mysql-connector==2.2.9
13. python-engineio==4.3.3
14. python-socketio==5.7.1
15. six==1.16.0
16. Werkzeug==2.2.2
17. zope.event==4.6
18. zope.interface==5.5.2

How can I fix this error?

**Comment by miguelgrinberg** [comment]
I think you have to pay more attention.

> it can happen due to multiple things like gevent eventlet. But I have not installed any one of these.

> eventlet==0.33.3

Why don't you try uninstalling `eventlet` and `greenlet` and see if that helps?

**Comment by PrimeTheFirst** [comment]
I did that. Its a no-go. Still invalid session. Its a 400 error with strict-origin-when-cross-origin Referrer Policy

**Reply by miguelgrinberg** [reply]
Instead of describing the errors that you get in your own words, please paste all exact copies of the logs in which these errors appear please.

**Comment by PrimeTheFirst** [comment]
App 3791622 output: Invalid session AGym7K7ze5H-mYvwAAAK (further occurrences of this error will be logged with level INFO)
App 3791622 output: [ pid=3791622, time=2023-01-28 20:05:51,048 ]: Invalid session AGym7K7ze5H-mYvwAAAK (further occurrences of this error will be logged with level INFO)
App 3791622 output: The WebSocket transport is not available, you must install a WebSocket server that is compatible with your async mode to enable it. See the documentation for details. (further occurrences of this error will be logged with level INFO)
App 3791622 output: [ pid=3791622, time=2023-01-28 20:05:51,321 ]: The WebSocket transport is not available, you must install a WebSocket server that is compatible with your async mode to enable it. See the documentation for details. (further occurrences of this error will be logged with level INFO)
App 3765393 output: received my event: {'data': 'User Connected'}
App 3791622 output: received my event: {'data': 'User Connected'}
App 3791622 output: received my event: {'data': 'User Connected'}
App 3791622 output: received my event: {'data': 'User Connected'}
App 3791622 output: received my event: {'data': 'User Connected'}
App 3791622 output: received my event: {'data': 'User Connected'}
App 3765393 output: received my event: {'data': 'User Connected'}
App 3791622 output: received my event: {'data': 'User Connected'}



Dev tools:
Failed to load resource: the server responded with a status of 400 (Bad Request) socket.io/EIO=4&transport=polling&t=ONvAkmn&sid=8dD42GnA5M224YGHAAAA


![image](https://user-images.githubusercontent.com/89402723/215277326-dd08ac28-5e1c-400b-85da-81ceafe31e45.png)

![image](https://user-images.githubusercontent.com/89402723/215277368-618e29ed-02ca-4c60-bfc2-647f5962acc8.png)

**Reply by miguelgrinberg** [reply]
> App 3791622 output: Invalid session AGym7K7ze5H-mYvwAAAK (further occurrences of this error will be logged with level INFO)

This is probably due to you running multiple workers with improper load balancing on your application. Switch to one multi-threaded worker, or else configure session affinity in your load balancer. See the deployment documentation for details.

> App 3791622 output: The WebSocket transport is not available, you must install a WebSocket server that is compatible with your async mode to enable it. See the documentation for details. (further occurrences of this error will be logged with level INFO)

To address this message, install simple-websocket:

    pip install simple-websocket
